### ISSUE TITLE
Multiple builders setting in setuptools does not work

### ISSUE THREAD

**Issue by ianychoi** [sphinx-doc/sphinx#4111]
### Problem
- According to https://github.com/sphinx-doc/sphinx/pull/3476 , Sphinx >= 1.6 supports multiple builders in setuptools. I followed the guidelines mentioned in the commit and builder documentation, but unfortunately executing setuptools with multiple builders returns an error.

#### Procedure to reproduce the problem
```
$ git clone git://git.openstack.org/openstack/horizon
$ cd horizon
(Adds "builder = html,latex" in setup.cfg - [build_sphinx] section)
$ tox -evenv -- python setup.py build_sphinx
```

#### Error logs / results
```
$ tox -evenv -- python setup.py build_sphinxvenv installed: alabaster==0.7.10,appdirs==1.4.3,asn1crypto==0.23.0,Babel==2.5.1,certifi==2017.7.27.1,cffi==1.11.0,chardet==3.0.4,cliff==2.9.0,cmd2==0.7.7,colorama==0.3.9,contextlib2==0.5.5,coverage==4.4.1,cryptography==2.0.3,debtcollector==1.18.0,Django==1.11.5,django-appconf==1.0.2,django-babel==0.6.1,django-compressor==2.2,django-nose==1.4.5,django-openstack-auth==3.5.0,django-pyscss==2.0.2,doc8==0.8.0,docutils==0.14,dulwich==0.18.3,enum34==1.1.6,extras==1.0.0,fasteners==0.14.1,fixtures==3.0.0,flake8==2.5.5,flake8-import-order==0.12,funcsigs==1.0.2,functools32==3.2.3.post2,futures==3.1.1,futurist==1.4.0,hacking==0.12.0,horizon==12.0.0.0rc2.dev138,idna==2.6,imagesize==0.7.1,ipaddress==1.0.18,iso8601==0.1.11,Jinja2==2.9.6,jsonpatch==1.16,jsonpointer==1.12,jsonschema==2.6.0,keystoneauth1==3.2.0,linecache2==1.0.0,MarkupSafe==1.0,mccabe==0.2.1,mock==2.0.0,monotonic==1.3,mox3==0.23.0,msgpack-python==0.4.8,netaddr==0.7.19,netifaces==0.10.6,nodeenv==1.2.0,nose==1.3.7,nose-exclude==0.5.0,nosehtmloutput==0.0.5,nosexcover==1.0.11,openstack.nose-plugin==0.11,openstackdocstheme==1.17.0,os-client-config==1.28.0,osc-lib==1.7.0,oslo.concurrency==3.23.0,oslo.config==4.13.0,oslo.i18n==3.18.0,oslo.policy==1.28.0,oslo.serialization==2.21.0,oslo.utils==3.30.0,osprofiler==1.13.0,pathlib==1.0.1,pbr==3.1.1,pep8==1.5.7,Pint==0.8.1,positional==1.2.1,prettytable==0.7.2,pycodestyle==2.3.1,pycparser==2.18,pyflakes==0.8.1,Pygments==2.2.0,pymongo==3.5.1,pyOpenSSL==17.3.0,pyparsing==2.2.0,pyperclip==1.5.27,pyScss==1.3.4,python-cinderclient==3.2.0,python-glanceclient==2.8.0,python-heatclient==1.11.0,python-keystoneclient==3.13.0,python-mimeparse==1.6.0,python-neutronclient==6.5.0,python-novaclient==9.1.0,python-swiftclient==3.4.0,pytz==2017.2,PyYAML==3.12,rcssmin==1.0.6,reno==2.5.0,requests==2.18.4,requestsexceptions==1.3.0,restructuredtext-lint==1.1.1,rfc3986==1.1.0,rjsmin==1.0.12,selenium==3.6.0,semantic-version==2.6.0,simplejson==3.11.1,six==1.11.0,snowballstemmer==1.2.1,Sphinx==1.6.4,sphinxcontrib-websupport==1.0.1,stevedore==1.27.0,termcolor==1.1.0,testtools==2.3.0,traceback2==1.4.0,typing==3.6.2,unicodecsv==0.14.1,unittest2==1.1.0,urllib3==1.22,warlock==1.2.0,WebOb==1.7.3,wrapt==1.10.11,XStatic==1.0.1,XStatic-Angular==1.5.8.0,XStatic-Angular-Bootstrap==2.2.0.0,XStatic-Angular-FileUpload==12.0.4.0,XStatic-Angular-Gettext==2.3.8.0,XStatic-Angular-lrdragndrop==1.0.2.2,XStatic-Angular-Schema-Form==0.8.13.0,XStatic-Bootstrap-Datepicker==1.3.1.0,XStatic-Bootstrap-SCSS==3.3.7.1,XStatic-bootswatch==3.3.7.0,XStatic-D3==3.5.17.0,XStatic-Font-Awesome==4.7.0.0,XStatic-Hogan==2.0.0.2,XStatic-Jasmine==2.4.1.1,XStatic-jQuery==1.10.2.1,XStatic-JQuery-Migrate==1.2.1.1,XStatic-jquery-ui==1.12.0.1,XStatic-JQuery.quicksearch==2.0.3.1,XStatic-JQuery.TableSorter==2.14.5.1,XStatic-JSEncrypt==2.3.1.1,XStatic-mdi==1.4.57.0,XStatic-objectpath==1.2.1.0,XStatic-Rickshaw==1.5.0.0,XStatic-roboto-fontface==0.5.0.0,XStatic-smart-table==1.4.13.2,XStatic-Spin==1.2.5.2,XStatic-term.js==0.0.7.0,XStatic-tv4==1.2.7.0,xvfbwrapper==0.2.9
venv runtests: PYTHONHASHSEED='768797199'
venv runtests: commands[0] | python setup.py build_sphinx
running build_sphinx
[pbr] Writing ChangeLog
[pbr] Generating ChangeLog
[pbr] ChangeLog complete (0.2s)
[pbr] Generating AUTHORS
[pbr] AUTHORS complete (0.6s)
Running Sphinx v1.6.4
connecting events for openstackdocstheme
loading pickled environment... done
Using openstackdocstheme Sphinx theme from /home/ian/git/openstack/horizon/.tox/venv/local/lib/python2.7/site-packages/openstackdocstheme/theme
building [mo]: all of 0 po files
building [html]: all source files
updating environment: 0 added, 0 changed, 0 removed
looking for now-outdated files... none found
preparing documents... done
writing output... [  1%] admin/admin-manage-roles
writing output... [  2%] admin/customize-configure
writing output... [  3%] admin/index
writing output... [  5%] admin/manage-flavors
writing output... [  6%] admin/manage-host-aggregates
writing output... [  7%] admin/manage-images
writing output... [  8%] admin/manage-instances
writing output... [ 10%] admin/manage-projects-and-users
writing output... [ 11%] admin/manage-services
writing output... [ 12%] admin/manage-shares
writing output... [ 13%] admin/manage-volumes
writing output... [ 15%] admin/sessions
writing output... [ 16%] admin/set-quotas
writing output... [ 17%] configuration/branding
writing output... [ 18%] configuration/customizing
writing output... [ 20%] configuration/index
writing output... [ 21%] configuration/pluggable_panels
writing output... [ 22%] configuration/settings
writing output... [ 23%] configuration/themes
writing output... [ 25%] contributor/contributing
writing output... [ 26%] contributor/faq
writing output... [ 27%] contributor/index
writing output... [ 28%] contributor/intro
writing output... [ 30%] contributor/quickstart
writing output... [ 31%] contributor/ref/context_processors
writing output... [ 32%] contributor/ref/decorators
writing output... [ 33%] contributor/ref/exceptions
writing output... [ 35%] contributor/ref/forms
writing output... [ 36%] contributor/ref/horizon
writing output... [ 37%] contributor/ref/index
writing output... [ 38%] contributor/ref/local_conf
writing output... [ 40%] contributor/ref/middleware
writing output... [ 41%] contributor/ref/tables
writing output... [ 42%] contributor/ref/tabs
writing output... [ 43%] contributor/ref/test
writing output... [ 45%] contributor/ref/workflows
writing output... [ 46%] contributor/testing
writing output... [ 47%] contributor/topics/angularjs
writing output... [ 48%] contributor/topics/index
writing output... [ 50%] contributor/topics/javascript_testing
writing output... [ 51%] contributor/topics/microversion_support
writing output... [ 52%] contributor/topics/packaging
writing output... [ 53%] contributor/topics/policy
writing output... [ 55%] contributor/topics/styling
writing output... [ 56%] contributor/topics/tables
writing output... [ 57%] contributor/topics/testing
writing output... [ 58%] contributor/topics/translation
writing output... [ 60%] contributor/topics/workflows
writing output... [ 61%] contributor/tutorials/dashboard
writing output... [ 62%] contributor/tutorials/index
writing output... [ 63%] contributor/tutorials/plugin
writing output... [ 65%] contributor/tutorials/table_actions
writing output... [ 66%] contributor/tutorials/workflow_extend
writing output... [ 67%] glossary
writing output... [ 68%] index
writing output... [ 70%] install/from-source
writing output... [ 71%] install/index
writing output... [ 72%] install/install-debian
writing output... [ 73%] install/install-obs
writing output... [ 75%] install/install-rdo
writing output... [ 76%] install/install-ubuntu
writing output... [ 77%] install/next-steps
writing output... [ 78%] install/plugin-registry
writing output... [ 80%] install/verify-debian
writing output... [ 81%] install/verify-obs
writing output... [ 82%] install/verify-rdo
writing output... [ 83%] install/verify-ubuntu
writing output... [ 85%] user/browser_support
writing output... [ 86%] user/configure-access-and-security-for-instances
writing output... [ 87%] user/create-networks
writing output... [ 88%] user/databases
writing output... [ 90%] user/index
writing output... [ 91%] user/launch-instances
writing output... [ 92%] user/log-in
writing output... [ 93%] user/manage-containers
writing output... [ 95%] user/manage-images
writing output... [ 96%] user/manage-lbaasv2
writing output... [ 97%] user/manage-shares
writing output... [ 98%] user/manage-volumes
writing output... [100%] user/stacks

generating indices... genindex py-modindex
highlighting module code... [  4%] horizon.forms.base
highlighting module code... [  9%] django.utils.functional
highlighting module code... [ 14%] horizon.tables.actions
highlighting module code... [ 19%] horizon.workflows.base
highlighting module code... [ 23%] horizon.middleware.base
highlighting module code... [ 28%] horizon.context_processors
highlighting module code... [ 33%] horizon.tabs.base
highlighting module code... [ 38%] horizon.decorators
highlighting module code... [ 42%] horizon.forms.fields
highlighting module code... [ 47%] horizon.forms.views
highlighting module code... [ 52%] horizon.tables.base
highlighting module code... [ 57%] horizon.exceptions
highlighting module code... [ 61%] horizon.workflows.views
highlighting module code... [ 66%] horizon.tabs.views
highlighting module code... [ 71%] openstack_dashboard.test.helpers
highlighting module code... [ 76%] horizon.tables.views
highlighting module code... [ 80%] horizon.tables.formset
highlighting module code... [ 85%] horizon.test.helpers
highlighting module code... [ 90%] django.utils.safestring
highlighting module code... [ 95%] horizon.base
highlighting module code... [100%] horizon.middleware.operation_log

writing additional pages... search
copying images... [  9%] user/figures/dashboard_project_tab.png
copying images... [ 18%] user/figures/create_image.png
copying images... [ 27%] user/figures/dashboard_admin_tab.png
copying images... [ 36%] contributor/topics/../../images/message_substitution.png
copying images... [ 45%] admin/figures/create_flavor.png
copying images... [ 54%] user/figures/dashboard_settings_tab.png
copying images... [ 63%] admin/figures/create_image.png
copying images... [ 72%] admin/figures/change_instance_state.png
copying images... [ 81%] user/figures/dashboard_identity_tab.png
copying images... [ 90%] contributor/topics/../../images/message_extraction.png
copying images... [100%] admin/figures/create_volume_type_encryption.png

copying static files... done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded.
Running Sphinx v1.6.4

Exception occurred:
  File "/home/ian/git/openstack/horizon/.tox/venv/local/lib/python2.7/site-packages/sphinx/util/logging.py", line 363, in filter
    raise SphinxWarning(message % record.args)
TypeError: not all arguments converted during string formatting
The full traceback has been saved in /tmp/sphinx-err-MdMomY.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
ERROR: InvocationError: '/home/ian/git/openstack/horizon/.tox/venv/bin/python setup.py build_sphinx'
___________________________________ summary ____________________________________
ERROR:   venv: commands failed

```

```
# Sphinx version: 1.6.4
# Python version: 2.7.12 (CPython)
# Docutils version: 0.14 
# Jinja2 version: 2.9.6
# Last messages:

# Loaded extensions:
Traceback (most recent call last):
  File "/home/ian/git/openstack/horizon/.tox/venv/local/lib/python2.7/site-packages/sphinx/setup_command.py", line 191, in run
    warningiserror=self.warning_is_error)
  File "/home/ian/git/openstack/horizon/.tox/venv/local/lib/python2.7/site-packages/sphinx/application.py", line 196, in __init__
    self.setup_extension(extension)
  File "/home/ian/git/openstack/horizon/.tox/venv/local/lib/python2.7/site-packages/sphinx/application.py", line 456, in setup_extension
    self.registry.load_extension(self, extname)
  File "/home/ian/git/openstack/horizon/.tox/venv/local/lib/python2.7/site-packages/sphinx/registry.py", line 207, in load_extension
    metadata = mod.setup(app)
  File "/home/ian/git/openstack/horizon/.tox/venv/local/lib/python2.7/site-packages/sphinx/ext/todo.py", line 231, in setup
    app.add_node(todolist)
  File "/home/ian/git/openstack/horizon/.tox/venv/local/lib/python2.7/site-packages/sphinx/application.py", line 530, in add_node
    type='app', subtype='add_node')
  File "/usr/lib/python2.7/logging/__init__.py", line 1456, in warning
    self.logger.warning(msg, *args, **kwargs)
  File "/usr/lib/python2.7/logging/__init__.py", line 1179, in warning
    self._log(WARNING, msg, args, **kwargs)
  File "/usr/lib/python2.7/logging/__init__.py", line 1286, in _log
    self.handle(record)
  File "/usr/lib/python2.7/logging/__init__.py", line 1296, in handle
    self.callHandlers(record)
  File "/usr/lib/python2.7/logging/__init__.py", line 1336, in callHandlers
    hdlr.handle(record)
  File "/usr/lib/python2.7/logging/__init__.py", line 755, in handle
    rv = self.filter(record)
  File "/usr/lib/python2.7/logging/__init__.py", line 617, in filter
    if not f.filter(record):
  File "/home/ian/git/openstack/horizon/.tox/venv/local/lib/python2.7/site-packages/sphinx/util/logging.py", line 363, in filter
    raise SphinxWarning(message % record.args)
TypeError: not all arguments converted during string formatting

```

#### Expected results
Outputs will be in both doc/build/html and doc/build/latex folders.

### Reproducible project / your project
- github.com/openstack/horizon

### Environment info
- OS: Ubuntu 16.04 LTS
- Python version: 2.7.12
- Sphinx version: 1.6.4
- venv installed: alabaster==0.7.10,appdirs==1.4.3,asn1crypto==0.23.0,Babel==2.5.1,certifi==2017.7.27.1,cffi==1.11.0,chardet==3.0.4,cliff==2.9.0,cmd2==0.7.7,colorama==0.3.9,contextlib2==0.5.5,coverage==4.4.1,cryptography==2.0.3,debtcollector==1.18.0,Django==1.11.5,django-appconf==1.0.2,django-babel==0.6.1,django-compressor==2.2,django-nose==1.4.5,django-openstack-auth==3.5.0,django-pyscss==2.0.2,doc8==0.8.0,docutils==0.14,dulwich==0.18.3,enum34==1.1.6,extras==1.0.0,fasteners==0.14.1,fixtures==3.0.0,flake8==2.5.5,flake8-import-order==0.12,funcsigs==1.0.2,functools32==3.2.3.post2,futures==3.1.1,futurist==1.4.0,hacking==0.12.0,horizon==12.0.0.0rc2.dev138,idna==2.6,imagesize==0.7.1,ipaddress==1.0.18,iso8601==0.1.11,Jinja2==2.9.6,jsonpatch==1.16,jsonpointer==1.12,jsonschema==2.6.0,keystoneauth1==3.2.0,linecache2==1.0.0,MarkupSafe==1.0,mccabe==0.2.1,mock==2.0.0,monotonic==1.3,mox3==0.23.0,msgpack-python==0.4.8,netaddr==0.7.19,netifaces==0.10.6,nodeenv==1.2.0,nose==1.3.7,nose-exclude==0.5.0,nosehtmloutput==0.0.5,nosexcover==1.0.11,openstack.nose-plugin==0.11,openstackdocstheme==1.17.0,os-client-config==1.28.0,osc-lib==1.7.0,oslo.concurrency==3.23.0,oslo.config==4.13.0,oslo.i18n==3.18.0,oslo.policy==1.28.0,oslo.serialization==2.21.0,oslo.utils==3.30.0,osprofiler==1.13.0,pathlib==1.0.1,pbr==3.1.1,pep8==1.5.7,Pint==0.8.1,positional==1.2.1,prettytable==0.7.2,pycodestyle==2.3.1,pycparser==2.18,pyflakes==0.8.1,Pygments==2.2.0,pymongo==3.5.1,pyOpenSSL==17.3.0,pyparsing==2.2.0,pyperclip==1.5.27,pyScss==1.3.4,python-cinderclient==3.2.0,python-glanceclient==2.8.0,python-heatclient==1.11.0,python-keystoneclient==3.13.0,python-mimeparse==1.6.0,python-neutronclient==6.5.0,python-novaclient==9.1.0,python-swiftclient==3.4.0,pytz==2017.2,PyYAML==3.12,rcssmin==1.0.6,reno==2.5.0,requests==2.18.4,requestsexceptions==1.3.0,restructuredtext-lint==1.1.1,rfc3986==1.1.0,rjsmin==1.0.12,selenium==3.6.0,semantic-version==2.6.0,simplejson==3.11.1,six==1.11.0,snowballstemmer==1.2.1,Sphinx==1.6.4,sphinxcontrib-websupport==1.0.1,stevedore==1.27.0,termcolor==1.1.0,testtools==2.3.0,traceback2==1.4.0,typing==3.6.2,unicodecsv==0.14.1,unittest2==1.1.0,urllib3==1.22,warlock==1.2.0,WebOb==1.7.3,wrapt==1.10.11,XStatic==1.0.1,XStatic-Angular==1.5.8.0,XStatic-Angular-Bootstrap==2.2.0.0,XStatic-Angular-FileUpload==12.0.4.0,XStatic-Angular-Gettext==2.3.8.0,XStatic-Angular-lrdragndrop==1.0.2.2,XStatic-Angular-Schema-Form==0.8.13.0,XStatic-Bootstrap-Datepicker==1.3.1.0,XStatic-Bootstrap-SCSS==3.3.7.1,XStatic-bootswatch==3.3.7.0,XStatic-D3==3.5.17.0,XStatic-Font-Awesome==4.7.0.0,XStatic-Hogan==2.0.0.2,XStatic-Jasmine==2.4.1.1,XStatic-jQuery==1.10.2.1,XStatic-JQuery-Migrate==1.2.1.1,XStatic-jquery-ui==1.12.0.1,XStatic-JQuery.quicksearch==2.0.3.1,XStatic-JQuery.TableSorter==2.14.5.1,XStatic-JSEncrypt==2.3.1.1,XStatic-mdi==1.4.57.0,XStatic-objectpath==1.2.1.0,XStatic-Rickshaw==1.5.0.0,XStatic-roboto-fontface==0.5.0.0,XStatic-smart-table==1.4.13.2,XStatic-Spin==1.2.5.2,XStatic-term.js==0.0.7.0,XStatic-tv4==1.2.7.0,xvfbwrapper==0.2.9

**Comment by tk0miya** [comment]
oh, sorry. This might be related with #4070.

**Comment by tk0miya** [comment]
Fixed at 50640b7.
Thank you for reporting
